core/arch/aarchxx/mangle.c

```diff
@@ -219,6 +219,99 @@ insert_clear_eflags(dcontext_t *dcontext, clean_call_info_t *cci,
      */
 }
 
+#ifdef AARCH64
+/* Creates a memory reference for registers saved/restored to memory. */
+static opnd_t
+create_base_disp_for_save_restore(uint base_reg, uint first_reg, uint reg,
+                                  bool is_single_reg, bool is_gpr)
+{
+
+    uint opsz;
+    if (is_gpr)
+        if (is_single_reg)
+            opsz = OPSZ_8;
+        else
+            opsz = OPSZ_16;
```

Comment:
I've added a comment at the top, stating that the opsz depends on the kind of register and if a single register or a pair of registers is saved/restored